target/mips/helper.c

```diff
@@ -100,9 +103,7 @@ int r4k_map_address(CPUMIPSState *env, hwaddr *physical, int *prot,
             }
 #if defined(TARGET_CHERI)
             if (rw == MMU_DATA_CAP_LOAD && (n ? tlb->L1 : tlb->L0)) {
-                env->TLB_L = 1;
-            } else {
-                env->TLB_L = 0;
+                *prot |= PAGE_LC_CLEAR;
```

Comment:
Presumably `prot` is only meant to be checked when we return `TLBRET_MATCH`; can we therefore move this down to where we have `*prot |= PAGE_WRITE;`, avoiding the `*prot = 0;` above and using `*prot = PAGE_READ;` from upstream again? Assuming my understanding is correct, of course...